### PR TITLE
EVG-7171 Set host throttle on admin page

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: units/crons.go [hunk 1]
```diff
@@ -842,7 +842,7 @@ func PopulateHostCreationJobs(env evergreen.Environment, part int) amboy.QueueOp
 			if h.UserHost || h.SpawnOptions.SpawnedByTask {
 				// pass:
 				//    always start spawn hosts asap
-			} else if submitted > 32 {
+			} else if submitted > env.Settings().HostInit.HostThrottle {
```
Comment (tychoish) [inline units/crons.go]: does .Settings() returned cached? I'm never clear on this.

Comment (tychoish) [reply units/crons.go]: its definitely cached, so to impact this you'd need to restart.

File: config_hostinit.go [hunk 2]
```diff
@@ -43,10 +46,16 @@ func (c *HostInitConfig) Set() error {
...
+			"host_throttle":    c.HostThrottle,
 		},
 	}, options.Update().SetUpsert(true))
 
 	return errors.Wrapf(err, "error updating section %s", c.SectionId())
 }
 
-func (c *HostInitConfig) ValidateAndDefault() error { return nil }
+func (c *HostInitConfig) ValidateAndDefault() error {
+	if c.HostThrottle == 0 {
```
Comment (ybrill) [inline config_hostinit.go]: should we check if `<= 0`?

File: config_hostinit.go [hunk 3]
```diff
@@ -43,10 +46,16 @@ func (c *HostInitConfig) Set() error {
 	_, err := coll.UpdateOne(ctx, byId(c.SectionId()), bson.M{
 		"$set": bson.M{
 			"ssh_timeout_secs": c.SSHTimeoutSeconds,
```
Comment (ybrill) [inline config_hostinit.go]: Many of these config subsections don't use `bsonutil.MustHaveTag` keys. 
Should we use them, though?

**Comment by john-m-liu** [comment]
Does this really fix anything? I would think that we would still have the same problems regardless of how much we throttle

**Comment by tychoish** [comment]
I think we want to do some empiricism with the size of the throttle, and making it configurable, is a good way to start that process.

File: config_hostinit.go (hunk 3 above)
Comment (bsamek) [reply config_hostinit.go]: I'm not sure. I followed the convention of the file. I think this is a thing we could go back and fix if we wanted to? @john-m-liu Any particular reason to leave `bsonutil.MustHaveTag` out here?

File: config_hostinit.go (hunk 2 above)
Comment (bsamek) [reply config_hostinit.go]: fixed

File: units/crons.go (hunk 1 above)
Comment (bsamek) [reply units/crons.go]: Is there a way to get fresh settings?

**Comment by bsamek** [comment]
> Does this really fix anything? I would think that we would still have the same problems regardless of how much we throttle

@john-m-liu It might not. There were three options in the ticket, and this is one of them. If it doesn't fix it, I'll move on to another option.

File: units/crons.go (hunk 1 above)
Comment (ybrill) [reply units/crons.go]: I think
https://github.com/evergreen-ci/evergreen/blob/7fd4857279b6263d3ae5b56d028777d2c94da5d2/config_hostinit.go#L17
will refresh from the db

File: config_hostinit.go (hunk 3 above)
Comment (john-m-liu) [reply config_hostinit.go]: It's the same reasoning as other places. Using it reduces the opportunity for programmer error but adds boilerplate/indirection

**Comment by bsamek** [comment]
@ybrill or @tychoish Could you take another look? https://github.com/evergreen-ci/evergreen/pull/3032/commits/dc2794a85f767d85d8b847db78efb856d90c8cd7 changes things just enough that this merits a re-look.

File: units/crons.go [hunk 4]
```diff
@@ -842,18 +842,23 @@ func PopulateHostCreationJobs(env evergreen.Environment, part int) amboy.QueueOp
...
-				// frequently, lets not start too many all at
-				// once.
-
-				continue
 			} else {
-				// only increment for task hosts, since otherwise
-				// spawn hosts and hosts spawned by tasks could
-				// starve task hosts
-				submitted++
+				config, err := evergreen.GetConfig()
```
Comment (john-m-liu) [inline units/crons.go]: Get the config outside of the loop

Comment (ybrill) [inline units/crons.go]: getting just the hostinit section could save db calls since we don't need all the configuration sections

Comment (bsamek) [reply units/crons.go]: fixed

Comment (bsamek) [reply units/crons.go]: fixed

**Pull request merged**